lib/ansible/module_utils/common/process.py

```diff
@@ -9,13 +9,13 @@
 from ansible.module_utils.common.file import is_executable
 
 
-def get_bin_path(arg, required=False, opt_dirs=None):
+def get_bin_path(arg, opt_dirs=None, required=None):
     '''
-    find system executable in PATH.
+    Find system executable in PATH. Raises ValueError if executable is not found.
     Optional arguments:
-       - required:  if executable is not found and required is true it produces an Exception
+       - required:  [Deprecated in 2.12] if executable is not found and required is true it produces an Exception
```

Comment:
The code implementing this has been removed.  So this docstring should be updated to say
```suggestion
       - required:  [Deprecated] Prior to 2.10, if executable is not found and required is true it produces an Exception.  In 2.10 and beyond, we always raise an exception, ignoring this parameter.  This parameter will be removed in 2.12.
```

(Change the version if applicable)

Alternately, the code should be changed to keep honoring the required parameter.  You might also want to use the deprecated functionality to mark this once you merge your code for doing that without access to an AnsibleModule.

Reply:
That would be good use of deprecations outside of an AnsibleModule object. I'll just update the doc string for now.